src/Number.php

```diff
@@ -115,4 +115,36 @@ public static function fromFloat($floatingPoint)
 
         return new self(sprintf('%.8g', $floatingPoint));
     }
+
+    /**
+     * @return bool
+     */
+    public function isNegative()
+    {
+        return $this->number[0] === '-';
+    }
+
+    /**
+     * @return string
+     */
+    public function getIntegerPart()
+    {
+        if ($this->decimalSeparatorPosition === false) {
+            return $this->number;
+        }
+
+        return substr($this->number, 0, $this->decimalSeparatorPosition);
+    }
+
+    /**
+     * @return string
+     */
+    public function getFractionalPart()
+    {
+        if ($this->decimalSeparatorPosition === false) {
+            return '';
```

Comment:
Or should I throw an exception here? My feeling is not: a number can have an empty fractional part.